pkg/config/setup/system_probe.go

```diff
@@ -379,6 +381,9 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 	// CCM config
 	cfg.BindEnvAndSetDefault(join(ccmNS, "enabled"), false)
 
+	// service_discovery module
+	cfg.BindEnvAndSetDefault(join(serviceDiscoveryNS, "enabled"), runtime.GOOS == "linux")
```

Comment:
ah ok, I guess this is the place where we set the default to enabled. Should the container check be moved here? Also, we should document this for when service discovery for windows is added in Q3.

Reply:
Please check @brycekahle previous review comments, I've followed the approach he recommended.
Makes sense, I can prepare some docs with all the necessary changes when working on the Windows implementation.